### PR TITLE
Added Readme and corresponding figure.

### LINKED ISSUE #44: Add a readme
For others to be able to use the code, please add some instructions in the readme how to run it :v: 🙏 

See e.g. here https://github.com/esa/NIDN or https://github.com/aidotse/PASEOS for examples (doesn't have to be quite that extensive)

### PR DESCRIPTION
**Pull request by rasmusmarak**
# Description

Summary of changes

* Updated the README.md file by adding new information.
* Added a docs/source folder than will in the future host logos, diagrams etc
* Added a diagram of the optimization scheme for the readme file to docs/source

## Related Pull Requests
- #58
- #54  

## Resolved Issues
- Fixes #44

### CODE REVIEW

**Comment by rasmusmarak** [comment]
**To be added in near future:**
- example notebook that give the user a detailed explanation of how to run the code for the 67P test case.
- notebook that displays how the optimization can be set up for other meshes, mission parameters and fitness functions.
- Further explanation and examples in readme about algorithm capabilities (such as possible numerical integrators, BFE capable optimization algorithms etc).

File: README.md [hunk 1]
```diff
@@ -1,6 +1,209 @@
...
+    <li><a href="#contact">Contact</a></li>
+  </ol>
+</details>
 
-The thesis topic was provided by the Advanced Concepts Team (ACT) at The European Space Agency (ESA).  
+<!-- ABOUT THE PROJECT -->
+
+## About The Project
+
+Trajectory Optimization for a Spacecraft Swarm (TOSS) is a thesis project presented for the degree of Master of Science in mathematics, with specialization in Optimization and Systems theory provided by KTH Royal Institute of Technology. The thesis topic was defined within the scope of $\Phi$[-lab@Sweden](https://www.ai.se/en/data-factory/f-lab-sweden) in the frame of a collaboration between [AI Sweden](https://www.ai.se/en/) and the [European Space Agency](https://www.esa.int/) to explore distributed edge learning for space applications. The project was mainly supervised by Pablo Gómez and Emmanuel Blazquez from the Advanced Concepts Team (ACT) at ESA, and examined by Xiaoming Hu from KTH.
```
Comment (gomezzz) [inline README.md:32]: I would move this into contact, most people won't care

File: README.md [hunk 2]
```diff
@@ -1,6 +1,209 @@
...
+
+- **open-source**: the source code of TOSS is publicly available.
+- **decentralised**: each function and capability within the module can be adopted to generate trajectories for a wider variety of objectives and celestial bodies.
+- **user-expandable**: the user can expand the capabilities of the code using the decentralised structure. 
+
+The optimization is based on an evolutionary approach sampling a large population where each candidate solution results in a deterministics trajectory that is obtained by numerical intergration. The trajectories are then resamples for a desired fixed time-step and then evaluating according to a defined fitness function. Then, unless the algorithm has met a certain stopping criteria, the population is mutated accoridng to the best candidate solution and the optimisation continuous by reiterating the process again.
+
+<p align="center">
+  <a href="https://github.com/rasmusmarak/TOSS">
+    <img src="docs/source/TOSS_Optimization_Scheme.pdf" alt="Scheme"  width="910" height="459">
```
Comment (gomezzz) [inline README.md:44]: doesn't render for me, can see online here https://github.com/rasmusmarak/TOSS/tree/Adding-Readme-to-repo

File: README.md [hunk 3]
```diff
@@ -1,6 +1,209 @@
...
+
+The goal of the project is to enable an evaluation of the effects for single and multi-spacecraft architectures when applied to a prospective mission with specified science objectives. In particular, TOSS is designed to be:
+
+- **open-source**: the source code of TOSS is publicly available.
+- **decentralised**: each function and capability within the module can be adopted to generate trajectories for a wider variety of objectives and celestial bodies.
+- **user-expandable**: the user can expand the capabilities of the code using the decentralised structure. 
+
+The optimization is based on an evolutionary approach sampling a large population where each candidate solution results in a deterministics trajectory that is obtained by numerical intergration. The trajectories are then resamples for a desired fixed time-step and then evaluating according to a defined fitness function. Then, unless the algorithm has met a certain stopping criteria, the population is mutated accoridng to the best candidate solution and the optimisation continuous by reiterating the process again.
```
Comment (gomezzz) [inline README.md:40]:
```suggestion
The optimization is based on an evolutionary approach sampling a large population where each candidate solution results in a deterministic trajectory that is obtained by numerical integration. The trajectories are then resamples for a desired fixed time-step and then evaluating according to a defined fitness function. Then, unless the algorithm has met a certain stopping criteria, the population is mutated according to the best candidate solution and the optimisation continuous by reiterating the process again.
```
would suggest to run a spellcheck on the file ;)

File: README.md [hunk 4]
```diff
@@ -1,6 +1,209 @@
...
+</details>
 
-The thesis topic was provided by the Advanced Concepts Team (ACT) at The European Space Agency (ESA).  
+<!-- ABOUT THE PROJECT -->
+
+## About The Project
+
+Trajectory Optimization for a Spacecraft Swarm (TOSS) is a thesis project presented for the degree of Master of Science in mathematics, with specialization in Optimization and Systems theory provided by KTH Royal Institute of Technology. The thesis topic was defined within the scope of $\Phi$[-lab@Sweden](https://www.ai.se/en/data-factory/f-lab-sweden) in the frame of a collaboration between [AI Sweden](https://www.ai.se/en/) and the [European Space Agency](https://www.esa.int/) to explore distributed edge learning for space applications. The project was mainly supervised by Pablo Gómez and Emmanuel Blazquez from the Advanced Concepts Team (ACT) at ESA, and examined by Xiaoming Hu from KTH.  
+
+The goal of the project is to enable an evaluation of the effects for single and multi-spacecraft architectures when applied to a prospective mission with specified science objectives. In particular, TOSS is designed to be:
```
Comment (gomezzz) [inline README.md:34]: Way too vague imo. You need to mention
- Optimizing trajectories with impulsive maneuvers around irregular bodies such as asteroids and comets using a polyhedral gravity model, RK xyz scheme and pygmo for evolutionary optimization.
- Goal of the trajectories is to cover the volume near the body well without getting too close

File: README.md [hunk 5]
```diff
@@ -1,6 +1,209 @@
...
+
+
+
+### Built With
+
+This project is based on:
+
+- [polyhedral-gravity-model](https://github.com/esa/polyhedral-gravity-model) a fast, parallelised version of the polyhedral gravity model implemented in C++ along with a python interface.
+- [DESolver](https://github.com/Microno95/desolver) a python library fro solving Initial Value Problems using numerical integrators. 
+- [PyGMo](https://esa.github.io/pygmo2/index.html) a scientific python library for massively parallel optimization managed by the Advanced Concepts Team at the European Space Agency
```
Comment (gomezzz) [inline README.md:63]: pygmo is C++ with a python wrapper I think

File: README.md [hunk 6]
```diff
@@ -1,6 +1,209 @@
...
+
+### Installation
+
+### conda
+The recommended way to install TOSS is via [conda](https://docs.conda.io/en/latest/) / [mamba](https://github.com/conda-forge/miniforge#mambaforge) using
+
+```
+conda install toss -c conda-forge
+
+```
```
Comment (gomezzz) [inline README.md:82]: ?? I don't think we are on conda

Comment (gomezzz) [reply README.md:82]: if you want to go in that direction I would suggest we open an issue to add pip first as that is way easier but would not write either here until it is actually possible

File: README.md [hunk 7]
```diff
@@ -1,6 +1,209 @@
...
+pytest
+```
+
+<!-- USAGE EXAMPLES -->
+
+## Usage
+
+### Config
+
+TOSS uses a central config file which is passed through the entire program. The default config parameters can be seen [here](https://github.com/rasmusmarak/TOSS/blob/2501351468eb362fa9ec8217b2fc426db37fca9a/toss/resources/default_cfg.toml).
```
Comment (gomezzz) [inline README.md:121]:
```suggestion
TOSS uses a central config file which is passed through the entire program. The default config parameters can be seen [here](https://github.com/rasmusmarak/TOSS/blob/main/toss/resources/default_cfg.toml). 
```
Why link to a specific commit?

File: README.md [hunk 8]
```diff
@@ -1,6 +1,209 @@
...
+
+<!-- USAGE EXAMPLES -->
+
+## Usage
+
+### Config
+
+TOSS uses a central config file which is passed through the entire program. The default config parameters can be seen [here](https://github.com/rasmusmarak/TOSS/blob/2501351468eb362fa9ec8217b2fc426db37fca9a/toss/resources/default_cfg.toml). 
+
+**[Coming Soon]** Practical usage of the config files will soon be added here through an explanatory notebook.
```
Comment (gomezzz) [inline README.md:123]: Maybe create an issue for it if you plan to do it? :)

File: README.md [hunk 9]
```diff
@@ -1,6 +1,209 @@
...
+
+### Use Case 1: Trajectory Optimization of a Swarm Orbiting 67P/Churyumov-Gerasimenko Maximising Gravitational Signal
+In this case, the aim is to compute a set of trajectories corresponding to a spacecraft swarm of n spacecraft, each with m impulsive maneuvers. The simulation is done within the context of a prospective mission around the comet 67P/Churyumov-Gerasimenko and the scientific target of maximising the measured gravitational signal. For details on mission parameters, optimization structure and fitness functions, see the corresponding paper on [ArXiv](https://doi.org/10.48550/arXiv.2306.01602).
```
Comment (gomezzz) [inline README.md:126]: I would suggest to use this https://citation-file-format.github.io/ , would also mention the paper higher up somewhere

File: README.md [hunk 10]
```diff
@@ -1,6 +1,209 @@
...
+
+To run this test case, install TOSS and setup the environment as presented above, then run the following:
+```
+OMP_NUM_THREADS=1 python toss.py
+```
+TOSS will then return the four csv files
```
Comment (gomezzz) [inline README.md:132]:
```suggestion
TOSS will then create four csv files
```

File: README.md [hunk 11]
```diff
@@ -1,6 +1,209 @@
...
+- **run_time**: which refers to the total simulation time.
+- **champion_f**: being the fitness value corresponding to the champion chromosome.
+- **champion_x**: being the champion chromosome corresponding to the optimal set of trajectories.
+- **fitness_list**: which is a list of champion fitness values correponding to each generation.
+
+### Use Case 2: Trajectory Optimization Around an Arbitrary Clestial Body Maximising Gravitation Signal
+**[Coming Soon]**
+
+### Use Case 3: Trajectory Optimization with User-Defined Scientific Objective
+**[Coming Soon]**
```
Comment (gomezzz) [inline README.md:142]: Issue for it?

File: README.md [hunk 12]
```diff
@@ -1,6 +1,209 @@
...
+and we will have a look at your contribution as soon as we can.
+
+Furthermore, please make sure that your Pull-Request passes all automated tests. Review will only happen after that.
+Only Pull-Requests created on the `main` branch with all tests passing will be considered.
+
+<!-- LICENSE -->
+
+## License
+
+**[Coming Soon]** Distributed under the .... License. See [LICENSE]() for more information.
```
Comment (gomezzz) [inline README.md:191]: need to update

**Review by gomezzz — CHANGES_REQUESTED** [review]
Some suggestions

File: README.md (hunk 1 above)
Comment (rasmusmarak) [reply README.md:32]: Fixed!

File: README.md (hunk 3 above)
Comment (rasmusmarak) [reply README.md:40]: Fixed!

File: README.md (hunk 4 above)
Comment (rasmusmarak) [reply README.md:34]: Fixed!

File: README.md (hunk 5 above)
Comment (rasmusmarak) [reply README.md:63]: Fixed!

File: README.md (hunk 7 above)
Comment (rasmusmarak) [reply README.md:121]: Fixed!

File: README.md (hunk 8 above)
Comment (rasmusmarak) [reply README.md:123]: There is one in #48 and I plan to do this as soon as possible, but likely not until right after the thesis has been officially finished.

File: README.md (hunk 9 above)
Comment (rasmusmarak) [reply README.md:126]: Added the cff-file in PR #54. Maybe we could merge that branch into this one for simplicity, and then I'll change the referencing in this PR directly before merging to main?

File: README.md (hunk 10 above)
Comment (rasmusmarak) [reply README.md:132]: Fixed!

File: README.md (hunk 11 above)
Comment (rasmusmarak) [reply README.md:142]: Added a new issue for this at #55.

File: README.md (hunk 6 above)
Comment (rasmusmarak) [reply README.md:82]: I removed this for now, but have added a new issue for this at #56.

File: README.md (hunk 12 above)
Comment (rasmusmarak) [reply README.md:191]: Referenced in issue #57 and added in PR #58.

**Comment by rasmusmarak** [comment]
# Note: 
The tests are failing due to polyhedral gravity model update v.2.0.0. Which is fixed in PR #46.

File: README.md (hunk 2 above)
Comment (rasmusmarak) [reply README.md]: Fixed for now, could perhaps be polished later on.

**Comment by rasmusmarak** [comment]
(just ignore commits ([478d076](https://github.com/rasmusmarak/TOSS/pull/47/commits/478d0766b50cd1899fc64b8e840cf0610ab515ee)) to ([4f1b26a](https://github.com/rasmusmarak/TOSS/pull/47/commits/4f1b26ac69b37d5ae777b4a32b2aad2d3c52fa84)). I found out that rendering pdf:s in readme was not really possible so these commits were mostly tests for that.)

File: README.md (hunk 9 above)
Comment (gomezzz) [reply README.md:117]: We can but it seems you branched out of main in #54 , thus it would be easier to merge them individually I think

**Review by gomezzz — APPROVED** [review]
:+1:

**Pull request merged**